src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final String TYPE_NAME_BULLDOZER = "Bulldozer";
+    private static final String TYPE_NAME_EXCAVATOR = "Excavator";
+    private static final String TYPE_NAME_Truck = "Truck";
+    private static final int COUNT_OF_FILL_LIST = 3;
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.getSimpleName().equals(TYPE_NAME_BULLDOZER)) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type.getSimpleName().equals(TYPE_NAME_EXCAVATOR)) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        } else if (type.getSimpleName().equals(TYPE_NAME_Truck)) {
+            return new ArrayList<>(new TruckProducer().get());
+        }
+        return new ArrayList<>();
```

Comment:
Isn't it better to return null?

Reply:
No, it's better to return empty list, because with null you may receive NPE